crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -0,0 +1,48 @@
+use tracing::debug;
+use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf};
+
+use crate::package_graph;
+
+pub struct PackageInference {
+    package_name: Option<String>,
+    directory_root: AnchoredSystemPathBuf,
+}
+
+impl PackageInference {
+    pub fn calculate(
+        turbo_root: &AbsoluteSystemPath,
+        pkg_inference_path: &AnchoredSystemPathBuf,
+        pkg_graph: &package_graph::PackageGraph,
+    ) -> Self {
+        debug!(
+            "Using {} as a basis for selecting pacakges",
+            pkg_inference_path
+        );
+        let full_inference_path = turbo_root.resolve(pkg_inference_path);
+        for (workspace_name, workspace_entry) in pkg_graph.workspaces() {
+            let pkg_path = turbo_root.resolve(&workspace_entry.package_json_path());
+            let inferred_path_is_below = pkg_path.contains(&full_inference_path);
```

Comment:
this is the same as saying the inferred path is the child of the package path, right?

Reply:
Yes